### DISCUSSION TITLE
Obtaining an instance of a class

### DISCUSSION THREAD

**Discussion by mkturtledude** [vfsfitvnm/frida-il2cpp-bridge#341] [Q&A] [answered]
The app I'm examining recently updated its Unity version to 2021.3.25 (not sure what was the previous one) and my scripts broke.

In particular, I had been using this function to identify an active instance of a class, given the class name and number of methods:

```
function findInstance(className : string, numberOfMethods : number) : Il2Cpp.Object {
    const im = Il2Cpp.Domain.assembly('Assembly-CSharp').image;
    const cls = im.classes
    let myClass = im.tryClass(className);
    for (const c of cls) {
        if (c.name == className && numberOfMethods == c.methods.length) {
            myClass = c;
        };
    }
    console.log("The selected class is ", myClass)

    const instances = Il2Cpp.GC.choose(myClass);
    console.log("Instances array:", instances);

    var classInstance: Il2Cpp.Object = null;
    Il2Cpp.GC.choose(myClass).forEach((instance: Il2Cpp.Object) => {
        classInstance = instance;
    });
    return classInstance;
}
```

Since the update, the first print works fine and I get the class with all its methods. But it crashes when `Il2Cpp.GC.choose(myClass)` is called.

Any ideas why or how I can debug it? I should say I'm using version 0.7.13 of frida-il2cpp-dump. I would be happy to upgrade if necessary, but even that version is supposed to support Unity 2021.3.25, so I'm not sure it will help. I'm a bit reluctant to upgrade since last time I had a hard time with the dependencies (If I recall correctly, I had to specify an old version of frida-compile), and this seems to be a more fundamental problem.

**Comment by commonuserlol** [comment]
just update to lastest verison and try again

**Reply by vfsfitvnm** [reply, marked as answer]
Yes, it was fixed in `v0.8.0`. I recommend upgrading to the latest version, the breaking changes should be manageable if you read and follow the releases (https://github.com/vfsfitvnm/frida-il2cpp-bridge/discussions/categories/announcements) in chronological order

**Reply by mkturtledude** [reply]
That worked, thanks!
